cmd/storage.go

```diff
@@ -37,20 +40,37 @@ var storageCreateCmd = &cobra.Command{
 	Example: `  # Create storage of size 1Gb to a component
   odo storage create mystorage --path=/opt/app-root/src/storage/ --size=1Gi
 	`,
-	Args: cobra.ExactArgs(1),
+	Args: cobra.MaximumNArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		client := getOcClient()
 		applicationName, err := application.GetCurrent(client)
 		checkError(err, "")
 		projectName := project.GetCurrent(client)
 		componentName := getComponent(client, storageComponent, applicationName, projectName)
+		currentComponentName, err := component.GetCurrent(client, applicationName, projectName)
+		var storageName string
+		if len(args) != 0 {
+			storageName = args[0]
+		} else {
+			storageName = currentComponentName + "-" + RandStringRunes()
 
-		_, err = storage.Create(client, args[0], storageSize, storagePath, componentName, applicationName)
+		}
+		_, err = storage.Create(client, storageName, storageSize, storagePath, componentName, applicationName)
 		checkError(err, "")
-		fmt.Printf("Added storage %v to %v\n", args[0], componentName)
+		fmt.Printf("Added storage %v to %v\n", storageName, componentName)
 	},
 }
 
+func RandStringRunes() string {
```

Comment:
There is a already a random string generator in pkg/uitl/util.go https://github.com/redhat-developer/odo/blob/0cc71d145bdc80fad2a9501d27e0904cc17735d5/pkg/util/util.go#L37

Can we reuse it?

Reply:
thanks @mik-dass , I didn't noticed :(